application/forms/AddToBasketForm.php

```diff
@@ -41,21 +38,14 @@ public function setup()
         $count = 0;
         $type = $this->type;
         foreach ($this->names as $name) {
-            if (! empty($names)) {
-                $names[] = ', ';
-            }
-            if ($basket && $basket->hasObject($type, $name)) {
-                $names[] = Html::tag('span', [
-                    'style' => 'text-decoration: line-through'
-                ], $name);
-            } else {
-                $count++;
+            if (! $basket || ! $basket->hasObject($type, $name)) {
                 $names[] = $name;
+                $count++;
             }
         }
         $this->addHtmlHint((new HtmlDocument())->add([
             'The following objects will be added: ',
-            $names
+            (array) implode(", ", $names)
```

Comment:
instead of casting the string to array could you not perform string concatenation?